Added Feature: splash screen

Srijani-Das07 opened the pull request:
# Pull Request

## Description

- Added SplashScreen with fade and scale animations
- Updated router to include splash screen as initial route
- Added spacing constants to AppConstants
- Fixed spacing widget usage in CreateEventScreen
- Integrated router in main.dart

Fixes #3

---

## Type of Change
- [x] Bug fix 🐛  
- [x] New feature ✨  
- [ ] Documentation update 📚  
- [ ] Refactor ♻️  
- [ ] Other (please describe):

---

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes.
- [ ] Unit tests
- [ ] Widget tests
- [x] Manual testing on device/emulator

---

## Screenshots (if applicable)
![Splash Screen](https://raw.githubusercontent.com/Srijani-Das07/CircleOfInterest/feature/splash_screen/assets/images/splash_screen.png)

---

## Checklist
- [x] My code follows the project’s style guidelines  
- [x] I have performed a self-review of my code  
- [x] I have commented my code where necessary  
- [ ] I have updated the documentation (if applicable)  
- [ ] I have added tests that prove my fix is effective or my feature works  
- [x] New and existing tests pass locally  

---

## Additional Notes
N/A

Srijani-Das07 committed:
feat: Added splash screen with animations and routing

Srijani-Das07 committed:
Added splash screen image

ktsrivastava29 commented:
Hey @Srijani-Das07 , thanks for the PR! 
About the screenshot you attached, I’m not sure which device it’s from. Could you please share one by running it on an actual Android or iOS device (a mobile phone specifically)?

Also, I noticed that all the splash screen code is currently in a single file. It would be better to move the animation-related code into a separate file and then import it into the splash screen. That way, the structure will be cleaner and easier to maintain.

Srijani-Das07 commented:
Hi @ktsrivastava29, Thank you for the feedback. I have run it locally on chrome right now. Sure, I'll run it on Android and share.
And also move the animation-related code into a separate file.

mahalaxmi-k commented:
hi @Srijani-Das07 are you still working on this issue? Also, don't forget to join our discord channel [https://discord.gg/WZemvWQdJy]

Srijani-Das07 commented:
@mahalaxmi-k , yes I'm still working on the issue. I'll send the pr as soon as possible.

Srijani-Das07 committed:
Created new file for splash screen animations

Srijani-Das07 committed:
Added image of splash_screen on android phone

Srijani-Das07 commented:
Hi @ktsrivastava29 and @mahalaxmi-k,
Sorry for the delay in getting back to you. I've made the requested changes:
- Moved the animation-related code into a separate file for better structure
- Tested and added a screenshot from a Android mobile device

Please review the updated PR. Let me know if any further changes are needed!

mahalaxmi-k commented:
okay. The admin will check it out and get back to you!

ktsrivastava29 commented:
@Srijani-Das07 Please remove all unnecessary comments from the code. Also, remove the assets you’ve added, there’s no need to include them in the codebase; you can simply share them here in the comments instead.

Additionally, there’s quite a bit of hard-coding in your implementation. Please review how this is typically handled in the existing codebase and update your changes accordingly.

Srijani-Das07 committed:
Updated splash_screen.dart and splash_animations.dart

Srijani-Das07 committed:
Added splash_constants.dart

Srijani-Das07 committed:
Removed demo images from assets

Srijani-Das07 committed:
Updated splash_screen.dart

Srijani-Das07 committed:
Updated splash_animations.dart
